drivers/interrupt_controller/intc_exti_stm32.c

```diff
@@ -0,0 +1,372 @@
+		__ASSERT_NO_MSG(0);								\
+	}
+
+
+bool stm32_exti_is_pending(uint32_t line_num)
+{
+	bool ret = false;
+	const uint32_t line = exti_linenum_to_ll_exti_line(line_num);
+
+	if (!IS_VALID_EXTI_LINE_NUM(line_num)) {
+		LOG_ERR("Invalid line number %u", line_num);
+		return false;
+	}
+
+	z_stm32_hsem_lock(CFG_HW_EXTI_SEMID, HSEM_LOCK_DEFAULT_RETRY);
+
+	/**
+	 * Note: we can't use EXTI_FN_HANDLER here because we care
+	 * about the return value of EXTI_IS_ACTIVE_FLAG.
+	 */
+	if (line_num < 32U) {
+		ret = EXTI_IS_ACTIVE_FLAG(0_31, line);
+#if STM32_EXTI_TOTAL_LINES_NUM > 32
+	} else if (line_num < 64U) {
+		ret = EXTI_IS_ACTIVE_FLAG(32_63, line);
+#endif /* STM32_EXTI_TOTAL_LINES_NUM > 32 */
+#if STM32_EXTI_TOTAL_LINES_NUM > 64
+	} else if (line_num < 96U) {
+		ret = EXTI_IS_ACTIVE_FLAG(64_95, line);
+#endif /* STM32_EXTI_TOTAL_LINES_NUM > 64 */
+	} else {
+		LOG_ERR("Invalid line number: %u", line_num);
+		__ASSERT_NO_MSG(0);
+	}
+
+	z_stm32_hsem_unlock(CFG_HW_EXTI_SEMID);
+
+	return ret;
+}
+
+int stm32_exti_clear_pending(uint32_t line_num)
+{
+	const uint32_t line = exti_linenum_to_ll_exti_line(line_num);
+
+	if (!IS_VALID_EXTI_LINE_NUM(line_num)) {
+		LOG_ERR("Invalid line number %u", line_num);
+		return -EINVAL;
+	}
+
+	z_stm32_hsem_lock(CFG_HW_EXTI_SEMID, HSEM_LOCK_DEFAULT_RETRY);
+
+	EXTI_FN_HANDLER(EXTI_CLEAR_FLAG, line_num, line);
+
+	z_stm32_hsem_unlock(CFG_HW_EXTI_SEMID);
+
+	return 0;
+}
+
+int stm32_exti_sw_interrupt(uint32_t line_num)
+{
+	const uint32_t line = exti_linenum_to_ll_exti_line(line_num);
+
+	if (!IS_VALID_EXTI_LINE_NUM(line_num)) {
+		LOG_ERR("Invalid line number %u", line_num);
+		return -EINVAL;
+	}
+
+	z_stm32_hsem_lock(CFG_HW_EXTI_SEMID, HSEM_LOCK_DEFAULT_RETRY);
+
+	EXTI_FN_HANDLER(EXTI_GENERATE_SWI, line_num, line);
+
+	z_stm32_hsem_unlock(CFG_HW_EXTI_SEMID);
+
+	return 0;
+}
+
+/** Enables the peripheral clock required to access EXTI registers */
+static int stm32_exti_enable_registers(void)
+{
+	/* Initialize to 0 for series where there is nothing to do. */
+	int ret = 0;
+#if defined(CONFIG_SOC_SERIES_STM32F2X) ||     \
+	defined(CONFIG_SOC_SERIES_STM32F3X) || \
+	defined(CONFIG_SOC_SERIES_STM32F4X) || \
+	defined(CONFIG_SOC_SERIES_STM32F7X) || \
+	defined(CONFIG_SOC_SERIES_STM32H7X) || \
+	defined(CONFIG_SOC_SERIES_STM32H7RSX) || \
+	defined(CONFIG_SOC_SERIES_STM32L1X) || \
+	defined(CONFIG_SOC_SERIES_STM32L4X) || \
+	defined(CONFIG_SOC_SERIES_STM32G4X)
+	const struct device *const clk = DEVICE_DT_GET(STM32_CLOCK_CONTROL_NODE);
+
+	if (!device_is_ready(clk)) {
+		LOG_ERR("Clock control device not ready");
+		return -ENODEV;
+	}
+
+	struct stm32_pclken pclken = {
+#if defined(CONFIG_SOC_SERIES_STM32H7X)
+		.bus = STM32_CLOCK_BUS_APB4,
+		.enr = LL_APB4_GRP1_PERIPH_SYSCFG
+#elif defined(CONFIG_SOC_SERIES_STM32H7RSX)
+		.bus = STM32_CLOCK_BUS_APB4,
+		.enr = LL_APB4_GRP1_PERIPH_SBS
+#else
+		.bus = STM32_CLOCK_BUS_APB2,
+		.enr = LL_APB2_GRP1_PERIPH_SYSCFG
+#endif /* CONFIG_SOC_SERIES_STM32H7X */
```

Comment:
Not required but as you're spending efforts to make things nice and right: It would be nice to make this depending on the presence of a `clocks` property.

Reply:
@I've added now `clocks = <&rcc STM32_CLOCK(APB4, 1U)>;` do h7 dts file and changed the function accordingly. Please take a look.